### ISSUE TITLE
Updating pnpm tries to write to a directory it shouldn't write to

### ISSUE THREAD

**Issue by miniBill** [pnpm/pnpm#3150]
### pnpm version:
5.11.1

### Code to reproduce the issue:
`pnpm i -g pnpm`

### Expected behavior:
`pnpm` gets updated

### Actual behavior:
```
 minibill@finland  ~  pnpm i -g pnpm
 ERROR  EROFS: read-only file system, open '/home/minibill/.nix-profile/bin/_tmp_20928_bf14b31111e7af2eb52c2438cf73b9e7'
For help, run: pnpm help add
 minibill@finland  ~  echo $PATH
/run/wrappers/bin:/home/minibill/.nix-profile/bin:/etc/profiles/per-user/minibill/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/home/minibill/.zsh/plugins/zsh-syntax-highlighting:/home/minibill/.zsh/plugins/zsh-autopair:/home/minibill/.zsh/plugins/powerlevel9k:/home/minibill/.zsh/plugins/nix-shell:/home/minibill/.npm/bin
 minibill@finland  ~  cat ~/.npmrc
prefix=/home/minibill/.npm
pnpm_prefix=/home/minibill/.npm
```

It is in fact true that `~/.nix-profile/bin` is readonly, but I see NO reason why `pnpm` should try and write there.

### Additional information:
 - `node -v` prints: v12.20.1
 - Windows, macOS, or Linux?: NixOS Linux

**Comment by zkochan** [comment]
This was fixed in pnpm v5.15 https://github.com/pnpm/pnpm/issues/2794

Try to install pnpm with one of the other methods: https://pnpm.js.org/en/installation

next time it will work

**Comment by miniBill** [comment]
This is not working on the latest pnpm version either
```
 minibill@finland  ~  pnpm --version
5.17.3
 minibill@finland  ~  pnpm i -g pnpm
 ERROR  EROFS: read-only file system, open '/home/minibill/.nix-profile/bin/_tmp_30498_39f825b0ef6eeb0a818be3f04b46707b'
For help, run: pnpm help add
```
